datafusion/physical-expr/src/expressions/get_indexed_field.rs

```diff
@@ -183,6 +186,14 @@ impl PhysicalExpr for GetIndexedFieldExpr {
         let array = self.arg.evaluate(batch)?.into_array(batch.num_rows());
         match &self.field {
             GetFieldAccessExpr::NamedStructField{name} => match (array.data_type(), name) {
+                (DataType::Map(_, _), ScalarValue::Utf8(Some(k))) => {
+                    let map_array = as_map_array(array.as_ref())?;
+                    let key_scalar = Scalar::new(StringArray::from(vec![k.clone()]));
+                    let keys = arrow_ord::cmp::eq(&key_scalar, map_array.keys())?;
```

Comment:
I think [`arrow::compute::eq`](https://docs.rs/arrow/latest/arrow/compute/kernels/cmp/fn.eq.html) is probably the more standard way to compare two arrays (though they call the same underlying kernels). That would let you avoid having to add the (newly explict) `arrow-ord` dependency.